packages/x-data-grid/src/hooks/features/sorting/gridSortingSelector.ts

```diff
@@ -73,3 +73,17 @@ export const gridSortColumnLookupSelector = createSelectorMemoized(
     return result;
   },
 );
+
+/**
+ * @category Sorting
+ * @ignore - do not document.
+ */
+export const gridSortedRowIndexLookupSelector = createSelectorMemoized(
+  gridSortedRowIdsSelector,
+  (sortedIds) => {
+    return sortedIds.reduce<Record<GridRowId, number>>((acc, id, index) => {
+      acc[id] = index;
+      return acc;
+    }, {});
```

Comment:
Nit: I think we could live without Object prototype with the selector to strip down some bytes of memory.
```suggestion
    }, Object.create(null));
```